apex/contrib/optimizers/distributed_fused_adam.py

```diff
@@ -223,12 +223,14 @@ def __init__(self,
                 f'(dtype={dtype}, '
                 f'grad_sync_dtype={grad_sync_dtype}, '
                 f'param_sync_dtype={param_sync_dtype}))')
-        if device != 'cuda':
-            raise RuntimeError('DistributedFusedAdam only supports GPU')
+        device = torch.device(device)
+        if (device.type != 'cuda'
+            or device.index not in (None, torch.cuda.current_device())):
+            raise RuntimeError(f'DistributedFusedAdam is only supported on current GPU')
```

Comment:
qq: would it be eqiuvalent to check all the params are a device tensor?

Reply:
We can actually handle params with any device or dtype, since we copy the values to the optimizer's internal tensors before calling any kernels. The real problem is that when I load a checkpoint, the device is `'cuda:{device_id}'` instead of `'cuda'`.